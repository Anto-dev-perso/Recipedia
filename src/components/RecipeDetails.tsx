

import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {Recipe} from '../utils/TableManipulation';

type RecipeDetailsProps = {
    recipe: Recipe;
    onClose: () => void;
  }

export default function RecipeDetails ({recipe, onClose}: RecipeDetailsProps) {
    return(
        <View>
            <Text style={{fontSize: 24, fontWeight: 'bold', marginBottom: 10}}>{recipe.title}</Text>
            <TouchableOpacity onPress={onClose}>
                <Text>Close</Text>
            </TouchableOpacity>
            <Text style={{fontSize: 18, fontWeight: 'bold', marginTop: 10}}>Ingredients:</Text>
            <Text style={{fontSize: 16, marginBottom: 10}}>{recipe.ingredients}</Text>
            <Text style={{fontSize: 18, fontWeight: 'bold', marginTop: 10}}>Instructions:</Text>
            <Text style={{fontSize: 16, marginBottom: 10}}>{recipe.instructions}</Text>
        </View>
    )
}