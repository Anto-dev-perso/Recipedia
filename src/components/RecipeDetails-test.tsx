/**
 * TODO fill this part
 * @format
 */

import React from "react";
import renderer from 'react-test-renderer';
import RecipeDetails from "./RecipeDetails";

// Integration test : example from Jtest with renderer from React. To consider also : Ract Native Testing Library

test('RecipeDetails component render correctly', () => {
    const tree = renderer.create(<RecipeDetails recipe={{title: "Test title", ingredients: "Test Ingredient 1, Test Ingredient 2", instructions: "Test Instruction 1, Test Instruction 2"}}  />).toJSON();
    expect(tree).toMatchSnapshot();
});