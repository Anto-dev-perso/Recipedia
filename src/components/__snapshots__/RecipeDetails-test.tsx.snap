// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails component render correctly 1`] = `
<View>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Test title
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Close
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    Ingredients:
  </Text>
  <Text
    style={
      {
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    Test Ingredient 1, Test Ingredient 2
  </Text>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    Instructions:
  </Text>
  <Text
    style={
      {
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    Test Instruction 1, Test Instruction 2
  </Text>
</View>
`;
